app.js

```diff
@@ -64,12 +64,12 @@ function save() {
 
 function load() {
   let parsedProducts = localStorage.getItem("productData");
-  JSON.parse(parsedProducts);
+  parsedProducts = JSON.parse(parsedProducts);
+  parsedProducts = allProducts;
 
-  for (let i = 0; i < allProducts.length; i++) {
-    let product = allProducts[i];
+  for (let i = 0; i < parsedProducts.length; i++) {
+    let product = parsedProducts[i];
     loadedProducts.push(product);
```

Comment:
There's no need for a for loop here if all you want to do is make a copy of the array you already have.